modules/services/ihaskell.nix

```diff
@@ -0,0 +1,113 @@
+# Adapted from Nixpkgs.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.ihaskell;
+  ihaskellSrc = pkgs.fetchFromGitHub {
```

Comment:
Why is this here? Is there no IHaskell package in Nixpkgs? If not then please add it to Nixpkgs and then refer to that package here. I'm very reluctant to add packages to Home Manager, they belong in Nixpkgs.

Reply:
ihaskell is marked as broken in nixpkgs. I'll try to open a PR over there. This is the only change left to be done in this PR. Thanks for your comments!

Reply:
If it helps, I pushed a new release to Hackage the other day: http://hackage.haskell.org/package/ihaskell-0.10.0.0, and I've unblocked it in Stackage: https://github.com/commercialhaskell/stackage/pull/4503

Reply:
Yep, everything looks quite nice now! I'd be happy to merge once the packaging is sorted. Thanks!

Reply:
@vaibhavsagar -  could you point me to the process for updating a haskell package (ihaskell here) in nixpkgs?

Reply:
It's pretty involved, I will open an issue :smile:.

Reply:
https://github.com/NixOS/nixpkgs/issues/60825